devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -387,11 +387,24 @@ private dimmerEvents(physicalgraph.zwave.Command cmd, ep = null) {
 	def value = (cmdValue ? "on" : "off")
 	def result = [createEvent(name: "switch", value: value)]
 	if (cmdValue && cmdValue <= 100) {
+		if(isDINDimmer()){
+			cmdValue = adjustValueToRange(cmd.value)
+		}
 		result << createEvent(name: "level", value: cmdValue == 99 ? 100 : cmdValue)
 	}
 	return result
 }
 
+Integer adjustValueToRange(value){
+	if(value == 0){
+		return 0
+	}
+	def minDimmingLvlPref = getCommandValue(parameterMap.find({it.key == 'minimumDimmingValue'})) ?: parameterMap.find({it.key == 'minimumDimmingValue'}).defaultValue
+	def range = 100 - minDimmingLvlPref
+	def adjustedValue = (((value-minDimmingLvlPref)*100)/range) as Integer
+	return Math.max(adjustedValue, minDimmingLvlPref)
```

Comment:
This math doesn't seem right to me.

If I want to set the level to 50, but my minimum value is 10:

range = 90
adjustedValue = ((40*100)/90) = 44.44

I would expect, by inputting 50, that I would get a value halfway between (50%) my minimum and maximum, which would be 55, right?

We want to treat [`minLevel`-100] as our range of values we will send to the device, but the value sent by the user should be interpreted as a percentage of that range. The way to get that number is `(100 - min) * level + min`.

Reply:
but retain the special case of 0 here.

Reply:
The issue here is that when I have:
minimum dimming value - 10
1.I chose to set 50% on UI and this value is being sent to the device
2.device responds that it set 55% - which is half of the range

I was recalculating those values to show 50 on UI in our application when creating ST events.

So it is: (55 - 10)*100/90

I see your point, but I think something is wrong 
(100 - min) * level + min = (100 - 10) * 55 + 10 = 4960

Reply:
Oh, I thought this was the code for sending the commands, not for parsing the reports. My mistake.

So if we send 50 to the device it reports 55, but we want to transform that back into 50, so we'd just do the above equation in reverse, solving for `value`: `(reportedValue - min)/(100 - min)`

`(55 - 10) / 90 = .5` so just multiply by 100 and boom, right?

Reply:
Ok, I changed this